Detectors/EMCAL/simulation/src/RawWriter.cxx

```diff
@@ -300,32 +301,21 @@ int RawWriter::carryOverMethod(const header::RDHAny* rdh, const gsl::span<char>
                                const char* ptr, int maxSize, int splitID,
                                std::vector<char>& trailer, std::vector<char>& header) const
 {
-  int offs = ptr - &data[0]; // offset wrt the head of the payload
-  // make sure ptr and end of the suggested block are within the payload
-  assert(offs >= 0 && size_t(offs + maxSize) <= data.size());
 
-  // Read trailer template from the end of payload
+  int bytesLeft = data.data() + data.size() - ptr;
+  int leftAfterSplit = bytesLeft - maxSize;
+
   gsl::span<const uint32_t> payloadwords(reinterpret_cast<const uint32_t*>(data.data()), data.size() / sizeof(uint32_t));
   auto rcutrailer = RCUTrailer::constructFromPayloadWords(payloadwords);
+  int trailerSize = rcutrailer.getTrailerSize() * sizeof(uint32_t);
```

Comment:
Isn't your trailed size constant?

Reply:
Yes, size is const at 9 32 bit words.

Reply:
then why do you need calculation of its size at every call?

Reply:
Yes, I am also more in favour of using a constant. Once we use a constant for number of words, shouldn't the calculation 
``` c++
int trailerSize = CONSTTRAILERSIZE * sizeof(uint32_t);
```
be constexpr?

Reply:
yes, just define ``constexpr int trailerSize = 9 * sizeof(uint32_t);``